network/omaha/server_test.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package omaha
+
+import (
+	"bytes"
+	"encoding/xml"
+	"io/ioutil"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/coreos/mantle/Godeps/_workspace/src/github.com/kylelemons/godebug/pretty"
+)
+
+func TestServerRequestResponse(t *testing.T) {
+	// make an omaha server
+	s, err := NewServer(":34567")
```

Comment:
Tests shouldn't use hard-coded ports, use ":0" and get the local address from the listener. So either NewServer should do this all the time and update Server.Addr or the api needs to be switched around to offer more control over the server's listener.